ui/packages/catalog-ui-search/src/main/webapp/react-component/utils/validation/validation.tsx

```diff
@@ -38,31 +51,275 @@ export function showErrorMessages(errors: any) {
 }
 
 export function getFilterErrors(filters: any) {
-  const errors: any[] = []
+  const errors = new Set()
+  let geometryErrors = new Set<string>()
   for (let i = 0; i < filters.length; i++) {
     const filter = filters[i]
-    const geometry = filter.geojson && filter.geojson.geometry
-    if (
-      geometry &&
-      geometry.type === 'Polygon' &&
-      geometry.coordinates[0].length < 4
-    ) {
-      errors.push({
-        title: 'Invalid geometry filter',
-        body:
-          'Polygon coordinates must be in the form [[x,y],[x,y],[x,y],[x,y], ... ]',
-      })
-    }
-    if (
-      geometry &&
-      geometry.type === 'LineString' &&
-      geometry.coordinates.length < 2
-    ) {
-      errors.push({
-        title: 'Invalid geometry filter',
-        body: 'Line coordinates must be in the form [[x,y],[x,y], ... ]',
-      })
-    }
+    getGeometryErrors(filter).forEach(function(msg) {
+      geometryErrors.add(msg)
+    })
+  }
+  geometryErrors.forEach(function(err) {
+    errors.add({
+      title: 'Invalid geometry filter',
+      body: err,
+    })
+  })
+  return Array.from(errors)
+}
+
+export function validateGeo(
+  key: string,
+  value: string,
+  value1?: any,
+  value2?: any,
+  value3?: any
+) {
+  switch (key) {
+    case 'lat':
+    case 'north':
+    case 'south':
+      return validateDDLatLon('latitude', 90, value)
```

Comment:
let's pull `latitude` & `longitude` into const strings at the top because I almost guarantee someone would ask you to do that when they review your PR 😆

Reply:
true true!! done